package/cpp/rnskia/RNSkDrawView.cpp

```diff
@@ -70,7 +70,7 @@ void RNSkDrawView::callJsDrawCallback(int width, int height, double timestamp) {
 
   // Set up arguments array
   std::vector<jsi::Value> args(2);
-  args[0] = jsi::Object::createFromHostObject(*runtime, _jsiCanvas);
```

Comment:
It shouldn't swap, and your fix is spot on correct! The reason for saving the canvas it in a class member (_jsiCanvas) is so that we don't need to create the _jsiWrapper between each call to render. Remember that our JsiHostObject classes have a caching mechanism that makes it a lot faster to call a method the second time, so it is a major difference performance wise to be able to keep the _jsiWrapper alive between calls.